src/main/java/data/Account.java

```diff
@@ -40,6 +44,21 @@ public Account(String accountID, String vendorID, String entity,
     this.aggregationMode = aggregationMode;
   }
 
+  public Account(HttpServletRequest request, String accountID, String vendorID) {
+    this.accountID = accountID;
+    this.vendorID = vendorID;
+    this.legacyAccountID = request.getParameter(BillingConfig.LEGACY_ACCOUNT_ID);
+    this.nextGenAccountID = Integer.parseInt(
+            request.getParameter(BillingConfig.NEXT_GEN_ACCOUNT_ID));
+    this.currency = request.getParameter(BillingConfig.CURRENCY_CODE);
+    this.direction = request.getParameter(BillingConfig.DIRECTION);
+    this.entity = request.getParameter(BillingConfig.ENTITY);
+    this.matchingMode = request.getParameter(BillingConfig.MATCHING_MODE);
+//    = request.getParameter(BillingConfig.PRODUCT_ACCOUNT_KEY);
+    // TODO: @Charlie should we throw this out?
```

Comment:
Yes, because you have the accountID field. We don't want to duplicate that field because product_account_key and accountID represent the same thing.

Reply:
check https://github.com/googleinterns/step57-2020/pull/16/commits/ea19a5d604973227e56b6a990b116db195938eab. I changed it so that it is still in the form and sets the account ID bc we don't have a different way of getting the account ID yet.